<!--Header-->
<div>
    <div class="row" style="color: #4D4D4D;font-size: 15px;padding-bottom: 20px">
        <div class="col-md-7">
            <img src="http://materials.cv.uoc.edu/cdocent/common/img/logo-uoc.png" alt="Logo UOC" class="img-responsive" style="margin:20px 0px 0px">
        </div>
        <div class="col-md-5">
            <h1 style="margin:15px 0px 0px;font-size: 40px;"></h1>
            <h2>Global Observatory on Donation and Transplantation</h2>
            <div style="text-align:left;margin-top: 5px;"></div>
        </div>
    </div>
    <div class="row" style="background: #FCB517;padding: 10px 20px;">
        <div class="col-md-6">
            <div style="text-align:right;">Autor: Jordi Puig Ovejero</div>
        </div>
    </div>
</div>
<!--/Header-->

## Índex ##
- [Selecció de les dades](#selecci%C3%B3-de-les-dades)
- [Relevància de les dades](#relev%C3%A0ncia-de-les-dades)
- [Diccionari de les variables](#diccionari-de-les-variables)
- [Dades Noves](#dades-noves)
- [Preguntes a respondre](#preguntes-a-respon)
- [Altres Visualitzacions](#altres-visualitzacions)


## Selecció de les dades ##

La meva selecció del conjunt de dades sobre els trasplantaments es basa en motius personals. Tinc familiars molt propers que han passat per aquest procés i sempre m'ha interessat entendre millor com funciona, quins són els factors que influeixen en l'èxit d'un trasplantament i com es distribueixen aquests procediments a nivell mundial. A més, crec que és un tema que pot generar consciència sobre la importància de la donació d'òrgans i pot ajudar a identificar àrees on es poden millorar les polítiques de salut. 

A nivell professional, com a estudiant de ciències de dades, aquest conjunt de dades presenta una oportunitat interessant per aplicar les habilitats que he après en l'anàlisi i visualització de dades. El conjunt de dades conté una gran quantitat d'informació que pot ser analitzada des de diferents perspectives, com ara per país, per tipus de trasplantament, per any, etc. Això permetrà desenvolupar visualitzacions de dades complexes i informatives. 

Finalment, crec que aquesta selecció és rellevant per a la societat en general, ja que els trasplantaments són procediments mèdics vitals que afecten a moltes persones arreu del món. Entendre millor aquesta temàtica pot contribuir a millorar la salut i el benestar de moltes persones.


## Rellevància de les dades ##

Les dades que utilitzo provenen de la font https://www.transplant-observatory.org/. Aquestes dades estan força actualitzades, ja que tenim registres fins a l'any 2023. Tot i que no hi ha molta informació històrica, ja que els registres més antics que tenim daten del 2000, són suficients per comprendre la situació del procés de trasplantament d'òrgans en els diferents països del món i quina incidència té.

## Descripció de les dades ##

#### Lectura del fitxer de dades ####

In [1]:
import numpy as np
import pandas as pd
import re as re
from scipy.stats import norm
import seaborn as sns

In [2]:
df = pd.read_csv('GODT.csv', header = 0)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3639 entries, 0 to 3638
Data columns (total 24 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   REGION              3639 non-null   object 
 1   COUNTRY             3639 non-null   object 
 2   REPORTYEAR          3639 non-null   int64  
 3   POPULATION          3639 non-null   object 
 4   TOTAL Actual DD     1758 non-null   float64
 5   Actual DBD          1645 non-null   float64
 6   Actual DCD          1418 non-null   float64
 7   Total Utilized DD   957 non-null    float64
 8   Utilized DBD        975 non-null    float64
 9   Utilized DCD        981 non-null    float64
 10  DD Kidney Tx        1780 non-null   float64
 11  LD Kidney Tx        1838 non-null   float64
 12  TOTAL Kidney Tx     1872 non-null   float64
 13  DD Liver Tx         1675 non-null   float64
 14  DOMINO Liver Tx     1318 non-null   float64
 15  LD Liver Tx         1552 non-null   float64
 16  TOTAL 

In [4]:
df.head()

REGION      COUNTRY  REPORTYEAR POPULATION  TOTAL Actual DD  \
0  Eastern Mediterranean  Afghanistan        2000       22,7              NaN   
1  Eastern Mediterranean  Afghanistan        2001       22,5              NaN   
2  Eastern Mediterranean  Afghanistan        2002       23,3              NaN   
3  Eastern Mediterranean  Afghanistan        2003       23,9              NaN   
4  Eastern Mediterranean  Afghanistan        2004       24,9              NaN   

   Actual DBD  Actual DCD  Total Utilized DD  Utilized DBD  Utilized DCD  ...  \
0         NaN         NaN                NaN           NaN           NaN  ...   
1         NaN         NaN                NaN           NaN           NaN  ...   
2         NaN         NaN                NaN           NaN           NaN  ...   
3         NaN         NaN                NaN           NaN           NaN  ...   
4         NaN         NaN                NaN           NaN           NaN  ...   

   DOMINO Liver Tx  LD Liver Tx  TOTAL Liver TX  Total Heart TX  DD Lung Tx  \
0              NaN          NaN             NaN             NaN         NaN   
1              NaN          NaN             NaN             NaN         NaN   
2              NaN          NaN             NaN             NaN         NaN   
3              NaN          NaN             NaN             NaN         NaN   
4              NaN          NaN             NaN             NaN         NaN   

   LD Lung Tx  TOTAL Lung Tx  Pancreas Tx  Kidney Pancreas Tx  Small Bowel Tx  
0         NaN            NaN          NaN                 NaN             NaN  
1         NaN            NaN          NaN                 NaN             NaN  
2         NaN            NaN          NaN                 NaN             NaN  
3         NaN            NaN          NaN                 NaN             NaN  
4         NaN            NaN          NaN                 NaN             NaN  

[5 rows x 24 columns]

El nostre conjunt de dades té 24 columnes o variables. El nombre total de registres és de 3639. 
Podem comprovar també que tenim un gran nombre de valors NaN. Haurem de veure com tractem aquests valors.    
També podem veure que tenim tant dades categòriques (REGION, COUNTRY o YEAR) i numèriques (tota la resta). YEAR la considerarem categòrica, no tindrà un significat numèric.

## Diccionari de les variables

#### `REGION`  
Dada categòrica (que podem fer servir com a geogràfica). Identifica la regió mundial del registre. Per exemple tenim Africa, Europe, Eastern Mediterranean...

#### `COUNTRY`  
Dada categòrica (també pot fer-se servir com a geogràfica). És el país del registre. 

#### `REPORTYEAR`  
Dada que encara que sigui un nombre, la farem servir com a categòrica. Any del registre.

**Nota: A partir d'ara totes les dades són numèriques.**

#### `POPULATION`  
Població del país en milions d'habitants

#### `TOTAL Actual DD`  
Total de donants difunts.

#### `Actual DBD`  
Total de donants difunts per mort cerebral (Deceased Donor Brain Dead). 

#### `Actual DCD`  
Total de donants difunts per aturada cardíaca (Deceased Donor Cardiac Death).

#### `Total Utilized DD`  
Total de donants difunts utilitzats.

#### `Utilized DBD`
Donants difunts per mort cerebral usats.

#### `Utilized DCD`
Donants difunts per aturada cardíaca usats.

#### `DD Kidney Tx`
Trasplantaments de ronyó de donants difunts.

#### `LD Kidney Tx`
Trasplantaments de ronyó de donants vius.

#### `TOTAL Kidney Tx`
Total de trasplantaments de ronyó

#### `DD Liver Tx`
Trasplantaments de fetge de donants difunts.

#### `DOMINO Liver Tx` 
Trasplantaments de fetge tipus domino. Dos receptors per un sol fetge.

#### `LD Liver Tx`  
Trasplantaments de fetge de donants vius.

#### `TOTAL Liver TX`  
Total de trasplantaments de fetge.

#### `Total Heart TX`  
Total de trasplantaments de cor.

#### `DD Lung Tx`  
Trasplantaments de pulmó de donants difunts.

#### `LD Lung Tx`  
Trasplantaments de pulmó de donants vius.

#### `TOTAL Lung Tx` 
Total de trasplantaments de pulmó.

#### `Pancreas Tx`  
Trasplantaments de pàncrees.

#### `Kidney Pancreas Tx`  
Trasplantaments de ronyó i pàncrees.

#### `Small Bowel Tx`  
Trasplantaments d'intestí prim.

## Noves dades:


#### `Transplant Rate per 100k`  
Total de trasplantaments de cor.

#### `Transplant Rate per 100k`  
Taxa de trasplantaments d'òrgans per cada 100.000 habitants a cada país o regió.

#### `Donor Ratio by Transplant Type`  
Proporció d'utilització de donants (sigui DBD o DCD) per a cada tipus de trasplantament d'òrgan (ronyó, fetge, cor, etc.).

#### `Total Utilized LD`  
Total de donants difunts utilitzats, sumant LD Kidney Tx, LD Liver Tx i LD Lung Tx

## Preguntes a respondre

Les preguntes que volem respondre en la o les visualitzacions:

1. **Distribució geogràfica dels trasplantaments**: que mostri la quantitat de trasplantaments d'òrgans diferents en diferents països o regions. Això ens ajudarà a identificar disparitats en l'accés als trasplantaments.

2. **Tendències temporals**: com ha variat el nombre de trasplantaments en el temps. Això ens permetrà identificar tendències al llarg dels anys.

3. **Proporció de tipus de donants i trasplantaments**: trobar la proporció de trasplantaments realitzats amb donants difunts per mort cerebral versus donants difunts per atur cardíac, així com la proporció de trasplantaments de diferents òrgans.

4. **Relació entre la població i el nombre de trasplantaments**: relació entre la població d'un país o regió i el nombre de trasplduts a termerealitzats en aquesta àrea.

5. **Comparació de diferents tipus de trasplantaments**: comparació entre la quantitat de trasplantaments de ronyó, fetge, cor, pduts a terme, realitzats en diferents països o regions.

6. **Eficiència en la utilització de donants**: comparació entre els totals de donants disponibles amb els trduts a terments realitzats per avaluar l'eficiència en la utilització dels òrgans donats en iferents països o regions.

## Altres visualitzacions:

Respecte d'aquestes dades, només he trobat algunes respostes de forma numèrica i visual a la pròpia web de [GODT](https://www.transplant-observatory.org/).

Per exemple podem veure de forma numèrica:

1. El nombre de trasplantaments totals del 2022.
2. L'increment respecte a l'any anterior.
3. Trasplantaments per hora al 2022.

De forma gràfica mitjançant un slide de Choropleth Maps:

1. Trasplantaments globals per regió.
2. Trasplantaments de cor, pulmó, ronyó... per regió.
3. Trasplantaments de donants vius per regió.


Les dades seran enriquides, però no amb noves dades sinó calculant altres dades a partir de les que tenim.

## Cita de les dades

* Those 2023 data are based on the Global Observatory on Donation and Transplantation (GODT) data, produced by the WHO-ONT collaboration.
* Data of the WHO-ONT Global Observatory on Donation and Transplantation.
* [Use of the Data](https://www.transplant-observatory.org/uses-of-dataquoting-data/)

<!--Footer-->
 <div style="background: #333333;padding: 35px 0px;margin-top: 25px;">
    <div class="row">
     <div class="col-sm-12">
        <img src="http://materials.cv.uoc.edu/cdocent/common/img/logo-uoc-bottom.png" alt="Logo UOC" class="img-responsive" style="margin: 0 auto; display: block;">
    </div>
</div>
</div>
<!--/Footer-->